### PR TITLE
Render polygon floorplan outlines in web tiles

### LINKED ISSUE #11300: web polygon floorplans are drawn as rectangles
### Describe the bug

In the web the outlines of the core and die are drawn as rectangles, but should honor the polygon outline.
Also the rows are not selectable as they are in the gui

Web:

Image

Gui:

Image

### Expected Behavior

Correct outlines

### Environment

```shell
HEAD of master
```

### To Reproduce

Load the ifp floorplans

### Relevant log output

```shell

```

### Screenshots

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by naveenvenk17**
Fixes #11300.

The web tile renderer was still drawing the die and core as bounding rectangles. That made non-rectangular floorplans look rectangular in the browser, even though the Qt viewer uses the actual polygon.

This uses the ODB die/core polygons and clips each edge to the tile before rasterizing it. I also added a pentagon regression test that checks the diagonal edge is present.

Tested on the GCP VM:
- /home/naveenvenkat/bin/bazel test --test_output=errors --cache_test_results=no //src/web/test:tile_generator_test
- /home/naveenvenkat/bin/bazel test --test_output=errors --cache_test_results=no //src/web/test:tile_generator_test //src/web/test:request_handler_test
- git diff --check

### CODE REVIEW

File: src/web/src/tile_generator.cpp [hunk 1]
```diff
@@ -3086,18 +3122,16 @@ std::vector<unsigned char> TileGenerator::renderTileBuffer(
       // on the _instances pass for all designs (Qt draws it always;
       // scoping to _instances keeps tech-layer tiles transparent).
       if (draw_die_outline || instances_only) {
-        const odb::Rect die = block->getDieArea();
-        if (die.area() > 0) {
-          outlineRectInTile(image_buffer, die, kOutlineGray, frame);
+        const odb::Polygon die = block->getDieAreaPolygon();
+        if (die.getEnclosingRect().area() > 0) {
+          outlinePolygonInTile(image_buffer, die, kOutlineGray, frame);
         }
```
[the remark this answers is not in the document]
Comment (gadfort) [reply src/web/src/tile_generator.cpp:3128]: I think this makes sense

File: src/web/src/tile_generator.cpp [hunk 2]
```diff
@@ -3086,18 +3122,16 @@ std::vector<unsigned char> TileGenerator::renderTileBuffer(
...
-      // Unset core -> empty polygon -> mergeInit()-inverted rect, so
-      // check min<max explicitly rather than area()>0.
       if (instances_only) {
-        const odb::Rect core = block->getCoreArea();
-        if (core.xMin() < core.xMax() && core.yMin() < core.yMax()) {
-          outlineRectInTile(image_buffer, core, kOutlineGray, frame);
+        const odb::Polygon core = block->getCoreAreaPolygon();
+        if (core.getEnclosingRect().area() > 0) {
+          outlinePolygonInTile(image_buffer, core, kOutlineGray, frame);
         }
```
[the remark this answers is not in the document]
Comment (gadfort) [reply src/web/src/tile_generator.cpp:3135]: I think this makes sense

**Review by gadfort — CHANGES_REQUESTED** [review]
LGTM, just fix the DCO issue and the code formatting. Thanks for working on this.

**Comment by naveenvenk17** [comment]
Thanks for the review. I addressed the formatting failures with clang-format 18, kept the empty-polygon check explicit, and squashed the branch into one signed commit for DCO.

I reran the full web C++ suite on the GCP VM:
- `bazel test --jobs=4 --test_output=errors --cache_test_results=no --runs_per_test=3 //src/web/test:cpp_tests`
- 4 tests passed three times
- `clang-format-18 --dry-run --Werror` on the changed C++ files
- `git diff --check`

**Review by gadfort — APPROVED** [review]
LGTM
